xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserPreferencesSheet.xml

```diff
@@ -247,6 +405,51 @@
   #end
 #end
 ###############################
+##   DISPLAY SHORTCUT PREFERENCE
+###############################
+#macro (displayShortcut $fieldName $label $translationKey)
+  #set ($escapedSelectId = $escapetool.xml("XWiki.UserPreferenceShortcutsClass_${fieldNumber}"))
+  #set ($escapedHintId = $escapedSelectId + "_hint")
+  #set ($defaultValue = $escapetool.xml($services.localization.render($translationKey)))
+  &lt;li&gt;
+    &lt;label #if($isEdit)for="$escapedSelectId"#end&gt;$!escapetool.xml($services.localization.render($label))
+      &lt;span id="$escapedHintId" class='xHint'&gt;##
+      $escapetool.xml($services.localization.render('userprofile.default')): $defaultValue##
+      &lt;/span&gt;
+    &lt;/label&gt;
+    #set ($shortcutPreference = '')
+    #set ($shortcutPreference = $doc.getObject('XWiki.UserPreferenceShortcutsClass', 'translationKey', $translationKey))
+    #set ($value = $doc.getValue('value', $shortcutPreference))
+    #if ($isEdit)
+      &lt;input type="text" class="shortcutPreference" name="$escapedSelectId" id="$escapedSelectId" size="30"
+        data-translatekey="$translationKey"
+        data-defaultvalue="$defaultValue"
+        #if ("$!shortcutPreference" != '')data-shortcutprefnumber='$shortcutPreference.getReference().getObjectNumber()'#end
+        #if ($services.localization.get("XWiki.XWikiUsers_${fieldName}.hint"))
+          #set ($escapedHintId = $escapedSelectId + "_hint")
+          aria-describedby="$escapedHintId"
+        #end
+        #if ($value)
+          value="$escapetool.xml($value)"
+        #elseif ($defaultValue)
+          value="$defaultValue"
+        #end&gt;
+      &lt;button class="btn btn-default reset-shortcut-preference" type="button&gt;
+        $services.icon.renderHTML('repeat') ##
+        $escapetool.xml($services.localization.render('platform.core.profile.shortcut.button.reset'))##
+      &lt;/button&gt;
+    #else
+      &lt;span&gt;
+        #if("$!shortcutPreference" != '') $!value##
+        #if("$!value" == '')&lt;small&gt;($escapetool.xml($services.localization.render('platform.core.profile.shortcut.hint.unbound')))&lt;/small&gt;#end
+        #else $defaultValue##
```

Comment:
To me, it looks wrong to have the `(` directly after the value, I suggest inserting a space before it:
```suggestion
        #else $defaultValue ##
```